src/AutoRest.CSharp/Output/Models/Types/OutputLibrary.cs

```diff
@@ -170,5 +177,95 @@ public RestClient FindRestClient(OperationGroup operationGroup)
             EnsureHeaderModels().TryGetValue(operation, out var model);
             return model;
         }
+
+        private void DecorateOperationGroup()
+        {
+            foreach (var operations in _codeModel.OperationGroups)
+            {
+                operations.ProviderName = _context.Configuration.OperationGroupMapping.ContainsKey(operations.Key) ? _context.Configuration.OperationGroupMapping[operations.Key] : ConstructOperationProviderName(operations);
+            }
+        }
+
+        private string ConstructOperationProviderName(OperationGroup operations)
+        {
+
+            string? providerName = "";
+            var request = GetBestMethod(operations);
+            bool adding = false;
+            if (request != null)
+            {
+                foreach (var segment in GetPathSegments(request.Path))
+                {
+                    var asSplit = segment.Split('/', StringSplitOptions.RemoveEmptyEntries);
+                    if (asSplit?.Length > 1 && asSplit.First().Equals("providers"))
+                    {
+                        adding = true;
+                        providerName = segment.Substring("providers".Length + 2).TrimEnd('/');
+                    }
+                    else if (adding)
+                    {
+                        providerName += segment.TrimEnd('/');
+                    }
+                }
+            }
+            return providerName.TrimEnd('/');
+        }
+        private static List<string> GetPathSegments(string httpRequestUri)
```

Comment:
Can https://github.com/Azure/autorest.csharp/blob/a853571978087e1ecd64c5d14c37887da4aee92e/src/AutoRest.CSharp/Utilities/StringExtensions.cs#L67 be used here?